src/lib/logSummary/logSummary.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+*
+* Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <unistd.h>
+#include <semaphore.h>
+#include <errno.h>
+#include <pthread.h>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "alarmMgr/alarmMgr.h"
+#include "logSummary/logSummary.h"
+
+
+
+/* ****************************************************************************
+*
+* Global vars
+*/
+static long long  transactionsAtLastSummary  = 0;
+static bool       logSummaryOn               = false;
+
+
+
+/* ****************************************************************************
+*
+* logSummary - 
+*/
+static void* logSummary(void* vP)
+{
+  int period = (long long) vP;
+
+  while (1)
+  {
+    long long transactionsNow;
+    long long diff;
+
+    sleep(period);
+
+    transactionsNow     = transactionIdGet();
+
+    // Has transactionId gone 'round-the-corner'?
+    if (transactionsNow < transactionsAtLastSummary)
+    {
+      diff = 0x7FFFFFFF - transactionsAtLastSummary + transactionsNow;
```

Comment:
transactionId is an int - 32 bits => 0x7FFFFFFF is max value (for signed int).
A long long (64 bits): 0x7FFFFFFFFFFFFFFF  (one '7' and 15 'F's)